### PR TITLE
Disable auto topic creation

### PR DESCRIPTION
**Pull request by kngu9**
We don't want to auto create topics if they don't exist.

### CODE REVIEW

File: charm/kafka/templates/server.properties [hunk 1]
```diff
@@ -41,6 +41,7 @@ ssl.client.auth=required
 # Hostname and port the broker will advertise to producers and consumers. If not set, 
 # it uses the value for "listeners" if configured.  Otherwise, it will use the value
 # returned from java.net.InetAddress.getCanonicalHostName().
+# NOTE: If we want to expose Kafka pubically, this has to be a public address or else Kafka will send the private ip address instead.
```
Comment (alesstimec) [inline charm/kafka/templates/server.properties]: s/publically/publicly/

**Review by alesstimec — APPROVED** [review]
LGTM with a nitpick

**Review by cmars — COMMENTED** [review]
Looks good, but per IRC chat, lets also:
- Add a charm config option to allow setting this to true. I think false is a sensible default though; prevents topics from getting created during a deployment before partitioning and replication have been explicitly set up.
- Manually QA test this with jmx-exporter and prometheus; confirm that the producer errors can be detected with our instrumentation. Just want to be sure we can detect and resolve a misconfiguration in production.

File: charm/bundle.yaml [hunk 2]
```diff
@@ -35,6 +35,9 @@ applications:
   nagios:
     charm: cs:nagios
     num_units: 1
+  jmx-exporter:
+    charm: /Users/khanh/Projects/jmx-exporter/builds/jmx-exporter
```
Comment (cmars) [inline charm/bundle.yaml]: This won't work for anyone else. Please use a charmstore URL.

File: charm/kafka/config.yaml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 options:
+  allow_topic_creation:
```
Comment (cmars) [inline charm/kafka/config.yaml]: Let's use the same language as the kafka config. `auto_create_topics` will be most familiar to kafka operators.

**Review by cmars — CHANGES_REQUESTED** [review]

**Comment by cmars** [comment]
@kngu9 This still needs fixing. I _think_ Kafka Streams creates the aggregate topics explicitly, so this is still a reasonable config option worth landing. The example bundle charm reference just needs fixing.

File: charm/kafka/templates/server.properties [hunk 4]
```diff
@@ -41,6 +41,7 @@ ssl.client.auth=required
 # Hostname and port the broker will advertise to producers and consumers. If not set, 
 # it uses the value for "listeners" if configured.  Otherwise, it will use the value
 # returned from java.net.InetAddress.getCanonicalHostName().
+# NOTE: If we want to expose Kafka publicly, this has to be a public address or else Kafka will send the private ip address instead.
```
Comment (cmars) [inline charm/kafka/templates/server.properties]: We don't want to expose Kafka publicly.

**Review by cmars — APPROVED** [review]
LGTM

**Pull request merged**